cmd/podman/shared/container.go

```diff
@@ -231,13 +232,27 @@ func NewBatchContainer(ctr *libpod.Container, opts PsOptions) (PsContainerOutput
 	pso.State = conState
 	pso.Pid = pid
 	pso.Size = size
-	pso.Pod = pod
 	pso.ExitedAt = exitedAt
 	pso.CreatedAt = ctr.CreatedTime()
 	pso.StartedAt = startedAt
 	pso.Labels = ctr.Labels()
 	pso.Mounts = strings.Join(ctr.UserVolumes(), " ")
 
+	if opts.Pod {
+		// The pod name is not in the container definition
+		// so we need to retrieve it using the pod ID.
+		var podName string
+		pod, err := r.LookupPod(podID)
```

Comment:
Potential improvement: We only need to do this if `podID` is not the empty string - saves us a database lookup, so should be a little faster

Reply:
Good idea! I've just made a change to improve this. Thanks again for your feedback.